src/Adapter/Product/CommandHandler/UpdateProductBasicInformationHandler.php

```diff
@@ -28,79 +28,101 @@
+        if (empty($updatableProperties)) {
             return;
         }
 
-        $this->performUpdate($product, CannotUpdateProductException::FAILED_UPDATE_BASIC_INFO);
+        $this->productRepository->partialUpdate($product, $updatableProperties, CannotUpdateProductException::FAILED_UPDATE_BASIC_INFO);
     }
 
     /**
      * @param Product $product
      * @param UpdateProductBasicInformationCommand $command
+     *
+     * @return array<string, mixed>
      */
-    private function fillUpdatableFieldsWithCommandData(
+    private function fillUpdatableProperties(
         Product $product,
         UpdateProductBasicInformationCommand $command
-    ): void {
+    ): array {
+        $updatableProperties = [];
+
         if (null !== $command->isVirtual()) {
             $product->is_virtual = $command->isVirtual();
-            $this->fieldsToUpdate['is_virtual'] = true;
+            $updatableProperties[] = 'is_virtual';
         }
 
         $localizedNames = $command->getLocalizedNames();
         if (null !== $localizedNames) {
             $product->name = $localizedNames;
-            $this->validateLocalizedField($product, 'name', ProductConstraintException::INVALID_NAME);
-            $this->setLocalizedFieldToUpdate('name', $localizedNames);
+            $updatableProperties['name'] = $this->getUpdatableLangs($localizedNames);
         }
 
         $localizedDescriptions = $command->getLocalizedDescriptions();
         if (null !== $localizedDescriptions) {
             $product->description = $localizedDescriptions;
-            $this->validateLocalizedField($product, 'description', ProductConstraintException::INVALID_DESCRIPTION);
-            $this->setLocalizedFieldToUpdate('description', $localizedDescriptions);
+            $updatableProperties['description'] = $this->getUpdatableLangs($localizedDescriptions);
         }
 
         $localizedShortDescriptions = $command->getLocalizedShortDescriptions();
         if (null !== $localizedShortDescriptions) {
             $product->description_short = $localizedShortDescriptions;
-            $this->validateLocalizedField($product, 'description_short', ProductConstraintException::INVALID_SHORT_DESCRIPTION);
-            $this->setLocalizedFieldToUpdate('description_short', $localizedShortDescriptions);
+            $updatableProperties['description_short'] = $this->getUpdatableLangs($localizedShortDescriptions);
         }
+
+        return $updatableProperties;
     }
 
     /**
-     * @param string $field
-     * @param array $localizedValues
+     * @param array $property
+     *
+     * @return int[]
      */
-    private function setLocalizedFieldToUpdate(string $field, array $localizedValues): void
+    private function getUpdatableLangs(array $property): array
     {
-        foreach ($localizedValues as $langId => $value) {
-            $this->fieldsToUpdate[$field][$langId] = true;
+        $updatableLangs = [];
+        foreach ($property as $langId => $value) {
```

Comment:
We might see this little logic duplicated in the future. When we see it we might refactor it.

Reply:
Yep, I agree we will meet this use case again I wonder if it couldn't be simplified with just using `array_keys($localizedDescriptions)` ?

Reply:
Using array keys would make all of langs updatable,  even those with null values.

Reply:
As discussed on slack. Its logical to use array_keys, because we wouldn't expect key=>value pairs in first place if it wasn't intentional to update it.